packages/imperative/src/events/src/EventUtils.ts

```diff
@@ -0,0 +1,199 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { ImperativeError } from "../../error/src/ImperativeError";
+import { join } from "path";
+import { ZoweUserEvents, ZoweSharedEvents, EventTypes, EventCallback } from "./EventConstants";
+import * as fs from "fs";
+import { ConfigUtils } from "../../config/src/ConfigUtils";
+import { IEventDisposable } from "./doc";
+import { Event } from "./Event";
+import { EventProcessor } from "./EventProcessor";
+
+/**
+ * A collection of helper functions related to event processing, including:
+ * - directory management
+ * - event type determination
+ * - subscription creation and setting callbacks
+ * - event writing
+ */
+export class EventUtils {
+
+    /**
+     * Retrieves a list of supported applications from configuration.
+     *
+     * @static
+     * @returns {string[]} List of application names.
+     */
+    public static getListOfApps(): string[] {
+        const extendersJson = ConfigUtils.readExtendersJsonFromDisk();
+        const apps: string[] = [];
+        // Loop through each profile type and accumulate all names and their sources based on conditions.
+        for (const [profileType, details] of Object.entries(extendersJson.profileTypes)) {
+            // Check each entry in the 'from' array to decide if a tag is needed
+            details.from.forEach(item => {
+                if (item.includes("(for VS Code)")) {
+                    apps.push(profileType, "_vsce"); // tag indicating Visual Studio Code Extension
+                } else if (item.includes("@zowe")) {
+                    apps.push(profileType); // no tag indicates Zowe CLI plugin (default)
+                }
+            });
+            apps.push("Zowe"); // default application name
+        }
+        return apps;
+    }
+
+    /**
+     * Won't throw an error if user entered a valid appName
+     *
+     * @static
+     * @param {string} appName - The name of the application.
+     */
+    public static validateAppName(appName: string){
+        const appList = this.getListOfApps();
+        if (appName !== "Zowe" && !appList.includes(appName)) {
```

Comment:
The `appName !== "Zowe"` check seems redundant since we always add `"Zowe"` to the list of app names generated by `this.getListOfApps`.

Reply:
good point 😋